Mage/src/main/java/mage/abilities/effects/common/cost/DynamicValueCostModificationEffect.java

```diff
@@ -0,0 +1,64 @@
+package mage.abilities.effects.common.cost;
+
+import mage.abilities.Ability;
+import mage.abilities.Mode;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.constants.CostModificationType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.game.Game;
+import mage.players.Player;
+import mage.util.CardUtil;
+
+import java.util.UUID;
+
+public class DynamicValueCostModificationEffect extends CostModificationEffectImpl {
```

Comment:
can apply this elsewhere later (such as in Pteramander)

Reply:
does this effect still apply if something else copies the ability?

Reply:
Hmm I'm not sure. I followed the pattern that other such cost reduction effects use of checking the "original ID". Not sure what the intended behavior needs to be in this case but it should be consistent with what is out there (e.g., Pteramander)

Reply:
Must follow name rules: `CostModificationSourceEffect`:
* no need in `DynamicValue` prefix;
* `SourceEffect` -- that's effect for source object and without target;
* BTW: `xxxTargetEffect` -- with targets; `xxxControlledEffect` for own objects only; `xxxAllEffect` -- for any objects.

Reply:
fixed

Reply:
I've got that @theelk801 talking about. `originalId` is smell code and can be broken for copied abilities/cards. See #5499 for more details and fixes example (compare source). Please, try that fix in PR.

Reply:
ah, gotcha, fixed following the example here: https://github.com/magefree/mage/commit/cae517e3105f9e3f2557c72dade71e6bedcc0f48